dev/io.openliberty.microprofile.reactive.messaging.internal_fat/fat/src/io/openliberty/microprofile/reactive/messaging/fat/kafka/metrics/MetricsTestServlet.java

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.microprofile.reactive.messaging.fat.kafka.metrics;
+
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.common.KafkaTestConstants;
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.framework.AbstractKafkaTestServlet;
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.framework.KafkaReader;
+
+import jakarta.inject.Inject;
+import jakarta.servlet.annotation.WebServlet;
+
+/**
+ *
+ */
+@SuppressWarnings("serial")
+@WebServlet("/MetricsTest")
+public class MetricsTestServlet extends AbstractKafkaTestServlet {
+
+    public static final String EMITTER_OUTGOING_CHANNEL = "MetricsEmitter";
+    public static final String EMITTER_TOPIC = "EmitterTopic";
+    public static final String OUTGOING_TOPIC = "OutgoingTopic";
+
+    @Inject
+    private MetricsDeliveryBean deliveryBean;
+
+    @Inject
+    @Channel(EMITTER_OUTGOING_CHANNEL)
+    private Emitter<String> emitter;
+
+    // called manually from MetricsTest
+    public void emitterDeliverPayload() {
+        // deliver 5 payloads via an emitter
+        for (int i = 1; i < 6; i++) {
+            emitter.send("Message " + i);
+        }
+
+        KafkaReader<String, String> reader = kafkaTestClient.readerFor(EMITTER_TOPIC);
+        reader.assertReadMessages(5, KafkaTestConstants.DEFAULT_KAFKA_TIMEOUT);
```

Comment:
As per the [other comment](https://github.com/OpenLiberty/open-liberty/pull/26640/files#r1365492208), replace this check with one that waits for the messages to be received by the reception bean.